### PR TITLE
Rename Hash to Root

### PR DESCRIPTION
**Pull request by terenc3t**
This PR addresses #285 with the following name changes:

- SSZTreeHash => hash_tree_root
- parent_hash => parent_root
- get_block_hash => get_block_root
- justified_block_hash => justified_block_root

### CODE REVIEW

File: specs/core/0_beacon-chain.md [hunk 1]
```diff
@@ -1292,9 +1292,9 @@ def exit_validator(state: BeaconState,
 Below are the processing steps that happen at every slot.
 
 * Let `latest_block` be the latest `BeaconBlock` that was processed in the chain.
-* Let `latest_hash` be the `SSZTreeHash` of `latest_block`.
+* Let `latest_hash` be the `hash_tree_root` of `latest_block`.
 * Set `state.slot += 1`
-* Set `state.latest_block_hashes = state.latest_block_hashes + [latest_hash]`. (The output of `get_block_hash` should not change, except that it will no longer throw for `state.slot - 1`).
+* Set `state.latest_block_hashes = state.latest_block_hashes + [latest_hash]`. (The output of `get_block_root` should not change, except that it will no longer throw for `state.slot - 1`).
```
Comment (djrtwo) [inline specs/core/0_beacon-chain.md]: Should change to `latest_block_roots`

File: specs/core/0_beacon-chain.md [hunk 2]
```diff
@@ -1292,9 +1292,9 @@ def exit_validator(state: BeaconState,
 Below are the processing steps that happen at every slot.
 
 * Let `latest_block` be the latest `BeaconBlock` that was processed in the chain.
-* Let `latest_hash` be the `SSZTreeHash` of `latest_block`.
+* Let `latest_hash` be the `hash_tree_root` of `latest_block`.
```
Comment (djrtwo) [inline specs/core/0_beacon-chain.md]: Rename `latest_root`

File: specs/core/0_beacon-chain.md [hunk 3]
```diff
@@ -1309,8 +1309,8 @@ If there is no block from the proposer at state.slot:
 
 ### Proposer signature
 
-* Let `block_hash_without_sig` be the `SSZTreeHash` of `block` where `block.signature` is set to `[0, 0]`.
-* Let `proposal_hash = SSZTreeHash(ProposalSignedData(state.slot, BEACON_CHAIN_SHARD_NUMBER, block_hash_without_sig))`.
+* Let `block_hash_without_sig` be the `hash_tree_root` of `block` where `block.signature` is set to `[0, 0]`.
+* Let `proposal_hash = hash_tree_root(ProposalSignedData(state.slot, BEACON_CHAIN_SHARD_NUMBER, block_hash_without_sig))`.
```
Comment (djrtwo) [inline specs/core/0_beacon-chain.md]: rename `proposal_root`

**Comment by terenc3t** [comment]
Thanks for the review @djrtwo @JustinDrake. I've updated all the fields except `ancestor_hashes` (getting pulled out)  Lmk if you want me to replace `ancestor_hashes` related fields too : )

File: specs/core/0_beacon-chain.md [hunk 4]
```diff
@@ -898,13 +898,13 @@ def get_beacon_proposer_index(state: BeaconState,
 
 #### `get_updated_ancestor_hashes`
 
-```python
+```pythonproposal_root
```
Comment (djrtwo) [inline specs/core/0_beacon-chain.md]: looks like something got messed up here
remove "proposal_root"

**Comment by JustinDrake** [comment]
Hey @terenc3t, if you want to do more cleanup PRs like this I'm happy to outsource a bunch :)

A few around functions which may catch your fancy:

* Put most function definitions in the "Helpers" section
* Also pull out functions like `get_ancestor` and `repeat_hash`
* Lint every code snippet with a Python style linter (ideally automating this process)
* Make sure every function definition has a typehints and docstrings
* Make sure the functions don't use hardcoded constants

**Pull request merged**